### DISCUSSION TITLE
To add a column with a default value to order by when adding a column

### DISCUSSION THREAD

**Discussion by rnrl1215** [ClickHouse/ClickHouse#57408] [Q&A]
ALTER TABLE test.table ADD COLUMN time_stamp DateTime64  DEFAULT now64(3, 'UTC'),
MODIFY ORDER BY (column1, column2, time_stamp )

I received message that newly added column ~~~ had a default expression 

When inserting data, I want to put the time stamp value with order by automatically

**Comment by den-crane** [comment]
DEFAULT expressions are forbidden in this case because they can make table corrupted. And you will get CANNOT_READ_ALL_DATA error.
The order of already stored rows can became inconsistent with the real order in case of ` DEFAULT now64`.

There is no good solution in this case. You should fill this column at a client.



You may try this as workaround https://fiddle.clickhouse.com/eed1d89f-5842-4ae1-aa48-9b64ce19d37b

```
alter table test add column d DateTime64(3), modify order by (a,d);

alter table test modify column d DateTime64(3) default 0; ---- puts 0 as default
alter table test materialize column d settings mutations_sync=2; --- materializes 0 in old data, need to wait until it finished
alter table test modify column d DateTime64(3) default now64(3); --- sets new default for the new data
```
